src/App.tsx

```diff
@@ -1,21 +1,118 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import './App.css';
 import { Header, GenreSelect, MovieGrid, MovieCounter } from './components';
+import { GENRES } from './constants';
+import SortControl from './components/SortControl/SortControl';
+import { MovieType } from './store/movies/types';
+import MovieDetails from './components/MovieDetails/MovieDetails';
+import Footer from './components/Footer/Footer';
+import UnderHeaderLine from './components/UnderHeaderLine/UnderHeaderLine';
+
+const genres = GENRES.filter((m) => m !== 'All');
+
+const randomGenerator = () => Math.floor(Math.random() * genres.length);
+
+const generateMovieGenres = () => {
+	let movieGenresLength = randomGenerator();
+	movieGenresLength = movieGenresLength === 0 ? 1 : movieGenresLength;
+	const genresList: string[] = [];
+	for (let i = 0; i < movieGenresLength; i++) {
+		const genre = genres[randomGenerator()];
+		if (!genresList.find((m) => m === genre)) {
+			genresList.push(genre);
+		}
+	}
+	return genresList;
+};
 
 function App() {
-	return (
-		<div className='container'>
-			<Header />
-			<GenreSelect
-				genres={['All', 'Documentary', 'Comedy', 'Horor', 'Crime', 'Drama']}
-				selectedGenre='All'
-				onSelect={(genre) => console.log('Selected genre:', genre)}
-			/>
-			<MovieCounter />
-			<MovieGrid />
-		</div>
-	);
+	const [name, setMovie] = useState(null);
+
+	const handleMovieClicked = (clickedMovieName?: string) => {
+		if (!clickedMovieName) {
+			setMovie(null);
+		} else {
+			setMovie(clickedMovieName);
```

Comment:
you can set whole movie object to this state and then pass it to Describe component.